codegen/MonocleCodegen.hs

```diff
@@ -0,0 +1,52 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+
+-- | A prototype to generate python dataclass from protobuf definition
+module Main (main) where
+
+import Language.ProtocolBuffers.Parser
+-- http://hackage.haskell.org/package/language-protobuf-1.0.1/docs/Language-ProtocolBuffers-Types.html
+import qualified Language.ProtocolBuffers.Types as PB
+import Relude
+
+-- | Create a python module from a protobuf
+-- >>> convertProtoToPython "../protos/config.proto"
+-- [ "@dataclass"
+-- , "class ProjectDefinition:"
+-- , "  name : str"
+-- , "  repository_regexp : str"
+-- , "  branch_regexp : str"
+-- , "  file_regexp : str"
+-- ]
+protoToPython :: PB.ProtoBuf -> [Text]
+protoToPython pb = concatMap pythonType (PB.types pb)
+  where
+    pythonType :: PB.TypeDeclaration -> [Text]
+    pythonType = \case
+      PB.DMessage name _ _ fields _ ->
+        [ "@dataclass",
+          "class " <> name <> ":"
+        ]
+          <> map pythonAttribute fields
+      _ -> []
+    pythonAttribute :: PB.MessageField -> Text
+    pythonAttribute = \case
+      PB.NormalField _rep pbType name _num _opts -> "  " <> name <> " : " <> pythonAttributeType pbType
+      _ -> ""
+    pythonAttributeType :: PB.FieldType -> Text
+    pythonAttributeType = \case
+      PB.TString -> "str"
+      _ -> "todo"
```

Comment:
Is there any existing library that handle that translate ?

Reply:
I have not found one, but that is very mechanical and it should be easy to render python or rescript type definition.

Reply:
Actually there is one, newest changes use the protobuf python library instead.